server/src/CompilationDatabase.cpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * \file
+ *
+ * \brief Using libclang for reading the JSON compilation database.
+ *
+ * This file is distributed under the GNU General Public License. See
+ * COPYING for details.
+ */
+#include <iostream>
+
+#include <clang-c/Index.h>
+#include <clang-c/CXCompilationDatabase.h>
+#include <clang-c/CXString.h>
+
+#include "CompilationDatabase.h"
+
+static void getExactFlags(std::vector<std::vector<std::string>> &flags,
+                          CXCompilationDatabase &db,
+                          const std::string &fullFilename) {
+  unsigned ncmd;
```

Comment:
In C++, unlike C we can create variables wherever we sit fit. It's better for code locality to declare them when used directly, at the smallest scope necessary.

So it should read:

```cpp
CXCompileCommands cmds = clang_CompilationDatabase_getCompileCommands(db, fullFilename.c_str());

for (unsigned i = 0, ncmd = clang_CompileCommands_getSize(cmds); i < ncmd; ++i) {
   // same inside the for-loop...
}
```